Add Gnome Shell 45 and 46 compatibility.

glerroo opened the pull request:
Since ESM files contain import and export keywords, extension modules won't be compatible with older GNOME Shell versions. The old shell versions must be removed and only use 45 in shell-version in metadata.json!

[Port Extensions to GNOME Shell 45](https://gjs.guide/extensions/upgrading/gnome-shell-45.html#shell-version)

This close #99 

![image](https://github.com/maoschanz/notes-extension-gnome/assets/73607126/110cab94-9ca6-4339-8bb5-f44d632fdd8e)

![image](https://github.com/maoschanz/notes-extension-gnome/assets/73607126/4419b41a-910c-4c97-bb2f-112387a37297)

glerroo committed:
Turn extension into module, use new convenience APIs & port to ESM all Gi/GS imports.

* Gnome Shell is moving to ESM, it will load extensions as standard modules.
* We can also start using standard imports for introspected modules & Gnome
Shell imports.
* Convenience APIs for extensions are now provided as Extension base classes.
* initTranslations()	method is considered deprecated.
* GLib timeout is to prefer instead of Mainloop.
* ByteArray.toString() is deprecated we must use TextDecoder.decode() instead.

glerroo committed:
Turn preferences into module, use new convenience APIs & port to ESM all Gi/GS imports.

* This is an initial working port, but no use adwaita widgets amd styles.

fewaltix commented:
Thank you so much for this commit, can confirm that it works flawlessly with Gnome 45 and OpenSuse Tumbleweed!

glerroo commented:
For me work great with GS45, but i don't use OpenSuse Tumbleweed, eventually i create a vm and test it.

FelipeFTN reviewed (approved):
**Great Work @glerroo! :rocket:**

fewaltix commented:
@glerroo 
I know! Just wanted to confirm it! Great work, thanks!

glerroo committed:
Fix a typo that prevents the "On the background" setting to working properly.

This fix issue #103.

glerroo committed:
Add Gnome Shell 46 support.

Clutter.Container was removed.
Clutter.Container.add_actor() and Clutter.Container.remove_actor() are
deprecated and should use Clutter.Actor.add_child() and
Clutter.Actor.remove_child() instead.

ppw0 commented:
Can you submit your version to extensions.gnome.org?

glerroo commented:
@maoschanz Hi, thanks for the extension, can you take a look to this pr and eventually merge it, and submit a new version to extensions.gnome.org.
